actors/common/src/main/java/org/sunbird/validator/user/UserBulkMigrationRequestValidator.java

```diff
@@ -0,0 +1,200 @@
+            }
+        });
+    }
+
+
+
+    private void csvRows(){
+        validateRowsCount();
+        migration.getValues().stream().forEach(migrationUser -> {
+            int index=migration.getValues().indexOf(migrationUser);
+            validateMigrationUser(migrationUser,index);
+        });
+        if(csvRowsErrors.getErrorsList().size()>0){
+            IErrorDispatcher errorDispatcher= ErrorDispatcherFactory.getErrorDispatcher(csvRowsErrors);
+            errorDispatcher.dispatchError();
+        }
+    }
+
+    private void validateRowsCount(){
+        if(migration.getValues().size()>=MAX_ROW_SUPPORTED){
+            throw new ProjectCommonException(
+                    ResponseCode.csvRowsExceeds.getErrorCode(),
+                    ResponseCode.csvRowsExceeds.getErrorMessage().concat("supported:"+MAX_ROW_SUPPORTED),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        }
+    }
+    private void validateMigrationUser(MigrationUser migrationUser,int index) {
+      emailAndPhone(migrationUser.getEmail(),migrationUser.getPhone(),index);
+      userExternalId(migrationUser.getUserExternalId(),index);
+      name(migrationUser.getName(),index);
+      orgExternalId(migrationUser.getOrgExternalId(),index);
+      channel(migrationUser.getChannel(),index);
+      inputStatus(migrationUser.getInputStatus(),index);
+    }
+    private void addErrorToList(ErrorDetails errorDetails){
+        if(errorDetails.getErrorEnum()!=null){
+            csvRowsErrors.setError(errorDetails);
+        }
+    }
+
+    public void emailAndPhone(String email, String phone,int index) {
+        ErrorDetails errorDetails=new ErrorDetails();
+        errorDetails.setRowId(index);
+        if(StringUtils.isBlank(email) && StringUtils.isBlank(phone)){
+            errorDetails.setErrorEnum(ErrorEnum.missing);
+            errorDetails.setHeader(JsonKey.EMAIL);
+        }
+        else if(StringUtils.isNotBlank(email)){
+            errorDetails.setHeader(JsonKey.EMAIL);
+            boolean isEmailValid=ProjectUtil.isEmailvalid(email);
+            if(!isEmailValid){
+            errorDetails.setErrorEnum(ErrorEnum.invalid);
+            }
+            if(!checkDuplicateValueOrAdd(emailSet,email)){
+                errorDetails.setErrorEnum(ErrorEnum.duplicate);
+            }
+
+        }
+        else  if(StringUtils.isNotBlank(phone)){
+            errorDetails.setHeader(JsonKey.PHONE);
+            boolean isPhoneValid=ProjectUtil.validatePhoneNumber(phone);
+            if(!isPhoneValid){
+                errorDetails.setErrorEnum(ErrorEnum.invalid);
+            }
+            if(!checkDuplicateValueOrAdd(phoneSet,phone)){
+                errorDetails.setErrorEnum(ErrorEnum.duplicate);
+            }
+        }
+        addErrorToList(errorDetails);
```

Comment:
we are always adding header name inside errorDetails. so make sure during display we are handling it.

Reply:
yes error is coming perfectly fine